plugins/techdocs-react/api-report.md

```diff
@@ -0,0 +1,116 @@
+## API Report File for "@backstage/plugin-techdocs-react"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+import { AsyncState } from 'react-use/lib/useAsync';
+import { ComponentType } from 'react';
+import { CompoundEntityRef } from '@backstage/catalog-model';
+import { Context } from 'react';
+import { Dispatch } from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { Extension } from '@backstage/core-plugin-api';
+import { default as React_2 } from 'react';
+import { SetStateAction } from 'react';
+import { VersionedValue } from '@backstage/version-bridge';
+
+// @alpha
+export function createTechDocsAddon<TComponentProps>(
+  options: TechDocsAddonOptions<TComponentProps>,
+): Extension<ComponentType<TComponentProps>>;
+
+// @alpha (undocumented)
+export const defaultTechDocsReaderPageValue: TechDocsReaderPageValue;
+
+// @alpha
+export const TECHDOCS_ADDONS_WRAPPER_KEY = 'techdocs.addons.wrapper.v1';
+
+// @alpha
+export enum TechDocsAddonLocations {
+  CONTENT = 'content',
+  HEADER = 'header',
+  PRIMARY_SIDEBAR = 'primary sidebar',
+  SECONDARY_SIDEBAR = 'secondary sidebar',
+  SUBHEADER = 'subheader',
+}
+
+// @alpha
+export type TechDocsAddonOptions<TAddonProps = {}> = {
+  name: string;
+  location: TechDocsAddonLocations;
+  component: ComponentType<TAddonProps>;
+};
+
+// @alpha
+export const TechDocsAddons: React_2.ComponentType;
+
+// @public
+export type TechDocsEntityMetadata = Entity & {
+  locationMetadata?: {
+    type: string;
+    target: string;
+  };
+};
+
+// @public
+export type TechDocsMetadata = {
+  site_name: string;
+  site_description: string;
+};
+
+// @alpha (undocumented)
+export const TechDocsReaderPageContext: Context<
+  | VersionedValue<{
+      1: TechDocsReaderPageValue;
+    }>
+  | undefined
+>;
+
+// @public
+export type TechDocsReaderPageValue = {
+  metadata: AsyncState<TechDocsMetadata>;
+  entityName: CompoundEntityRef;
```

Comment:
I think let's stick to `entityRef` for both `string` and `CompoundEntityRef` types, and `entity` for the `Entity` type. It's what we've settled on in most other parts. A pattern we use in some places is `entityRef: string | CompoundEntityRef` and then parse it first thing

Reply:
Hmm...  I think this ended up this way because we duplicated a type/pattern from the main TechDocs plugin (which still exists).  I _think_ we could change that here without a problem (and perhaps also just remove the deprecated `onReady` prop at the same time).

Will take a look.